deeplearning/Dockerfile

```diff
@@ -27,19 +23,35 @@ RUN apt-get update -qq && DEBIAN_FRONTEND=noninteractive apt-get install -yq --n
 		xterm \
 		&& rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
-# Create jovyan user with UID=1000 and in the 'users' group
-RUN groupadd -g 1019 bethgelab && \
-	groupadd -g 1011 cin && \
-	useradd -m -s /bin/bash -N -u $NB_UID $NB_USER && \
-	usermod -g bethgelab $NB_USER && \
-	usermod -a -G cin $NB_USER && \
-	usermod -a -G sudo $NB_USER
-
 # Enable passwordless sudo for users under the "sudo" group
 RUN sed -i.bkp -e \
       's/%sudo\s\+ALL=(ALL\(:ALL\)\?)\s\+ALL/%sudo ALL=NOPASSWD:ALL/g' \
 	  /etc/sudoers
 
+# Setup gosu (https://github.com/tianon/gosu)
+# gosu is an improved version of su which behaves better inside docker
+# we use it to dynamically switch to the desired user in the entrypoint
+# (see below)
+ENV GOSU_VERSION 1.9
+# Use unsecure HTTP via Port 80 to fetch key due to firewall in CIN.
+RUN set -x \
```

Comment:
Many lines for a simple program ;-). Can we leave the verification or use apt-get (e.g. https://launchpad.net/~tianon/+archive/ubuntu/gosu)?